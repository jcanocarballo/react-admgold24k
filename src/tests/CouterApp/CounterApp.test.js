const { default: CouterApp } = require("../../CouterApp/CouterApp");
import {shallow} from 'enzyme'

describe('Prueba del componente CouterApp', () => {
  let wraper = shallow(<CouterApp></CouterApp>);
  beforeEach( () => {
    wraper = shallow(<CouterApp></CouterApp>);
  });

  test('debe mostrar <CounterApp /> correctamente ', () => {    
    expect(wraper).toMatchSnapshot();    
  });
  
  test('debe mostrar el valor por defecto de 100', () => {
    const wraper = shallow(
      <CouterApp value={100}></CouterApp>
    );

    const counterText = wraper.find('h2').text().trim();
    expect(counterText).toBe('100');
  });

  test('debe incrementar con el boton add', () => {
    wraper.find('button').at(0).simulate('click');
    const counterText = wraper.find('h2').text().trim();
    expect(counterText).toBe('11');
  });
  
  test('debe colocar el valor inicial con el boton reset', () => {
    const wraper = shallow(
      <CouterApp value={105}></CouterApp>
    );
    wraper.find('button').at(0).simulate('click');
    wraper.find('button').at(0).simulate('click');
    wraper.find('button').at(1).simulate('click');
    const counterText = wraper.find('h2').text().trim();
    expect(counterText).toBe('105');
  });

  test('debe decrementar con el boton remove', () => {    
    wraper.find('button').at(2).simulate('click');
    const counterText = wraper.find('h2').text().trim();
    expect(counterText).toBe('9');
  });
});