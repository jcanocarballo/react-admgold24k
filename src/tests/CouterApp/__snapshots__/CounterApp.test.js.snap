// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente CouterApp debe mostrar <CounterApp /> correctamente  1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    ADD
  </button>
  <button
    onClick={[Function]}
  >
    RESET
  </button>
  <button
    onClick={[Function]}
  >
    REMOVE
  </button>
</div>
`;
